src/gss_names.c

```diff
@@ -281,6 +281,84 @@ uint32_t gssntlm_import_name(uint32_t *minor_status,
                                        output_name);
 }
 
+size_t gssntlm_get_attrs_count(const struct gssntlm_name_attribute *attrs)
+{
+    size_t attrs_count = 0;
+    if (!attrs) {
+        return 0;
+    }
+    while (attrs[attrs_count].attr_name != NULL) {
+        attrs_count++;
+    }
+
+    return attrs_count;
+}
+
+int gssntlm_copy_attrs(const struct gssntlm_name_attribute *src,
+                       struct gssntlm_name_attribute **dst)
+{
+    struct gssntlm_name_attribute *copied_attrs;
+    size_t attrs_count = gssntlm_get_attrs_count(src);
+
+    if (attrs_count == 0) {
+        *dst = NULL;
```

Comment:
Ok, I was surprised by the fact that we need to set OUTPUT dst value to NULL if we return an error code (usually in case of the error all OUTPUT parameters are untouched). If you have this convention - let it be...

Reply:
This is the convention in MIT krb5 code, the reason is that if we set all outputs at the start we do not risk a code path where we forget to set one, it is just defensive programming.

Reply:
Ok, good rationale